### PR TITLE
AMD MSR 

### PR DESCRIPTION
**Pull request by jaysongiroux**

### CODE REVIEW

File: data/amd/register/msr/MSRC000_0082.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+- name: MSRC000_0082
+  long_name: "SYSCALL Target Address"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC000_0082.yml:2]: This should be `long_name: "Long Mode SYSCALL Target Address"`

File: data/amd/register/msr/MSRC000_0084.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+- name: MSRC000_0084
+  long_name: "Compatibility Mode SYSCALL Target Address"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC000_0084.yml:2]: This should be `long_name: "SYSCALL Flag Mask"`

File: data/amd/register/msr/MSRC000_00E7.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+- name: MSRC000_00E7
+  long_name: "Read-Only Max Performance Frequency Clock Count"
+  purpose: |
+       "
+        MSRC000_0084 [SYSCALL Flag Mask] (Core::X86::Msr::SYSCALL_FLAG_MASK)
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC000_00E7.yml:5]: This purpose looks wrong

File: data/amd/register/msr/MSRC000_00E8.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+- name: MSRC000_00E8
+  long_name: "Read-Only Max Performance Frequency Clock Count"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC000_00E8.yml:2]: This should be `long_name: "Read-Only Actual Performance Frequency Clock Count"`

File: data/amd/register/msr/MSRC000_00E8.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  purpose: |
+       "
+        MSRC000_0084 [SYSCALL Flag Mask] (Core::X86::Msr::SYSCALL_FLAG_MASK)
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC000_00E8.yml:5]: This purpose is wrong

File: data/amd/register/msr/MSRC000_0101.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+- name: MSRC000_0101
+  long_name: "FS Base"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC000_0101.yml:2]: This should be `long_name: "GS Base"`

File: data/amd/register/msr/MSRC000_0410.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+- name: MSRC000_0410
+  long_name: "FS Base"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC000_0410.yml:2]: This should be `long_name: "MCA Interrupt Configuration"`

File: data/amd/register/msr/MSRC001_0015.yml [hunk 8]
```diff
@@ -0,0 +1,151 @@
+- name: MSRC001_0015
+  long_name: "System Configuration"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0015.yml:2]: This should be `long_name: "Hardware Configuration"`

File: data/amd/register/msr/MSRC001_0016.yml [hunk 9]
```diff
@@ -0,0 +1,54 @@
+- name: MSRC001_0016
+  long_name: "System Configuration"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0016.yml:2]: This should be 'long_name: "IO Range Base"'

File: data/amd/register/msr/MSRC001_0017.yml [hunk 10]
```diff
@@ -0,0 +1,38 @@
+- name: MSRC001_0017
+  long_name: "System Configuration"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0017.yml:2]: This should be: 'long_name: "IO Range Mask"'

File: data/amd/register/msr/MSRC001_0055.yml [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+       "
+  size: 64
+  arch: amd64
+
+  access_mechanisms:
+      - name: rdmsr
+        is_read: True
+        address: 0xC0010055
+
+      - name: wrmsr
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0055.yml:15]: The description of this register says "Read Only", so delete the wrmsr access mechanism

File: data/amd/register/msr/MSRC001_0055.yml [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+        is_write: True
+        address: 0xC0010055 
+
+  fieldsets:
+      - name: latest
+        condition: "Fieldset valid on latest version of the AMD architecture"
+        size: 64
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0055.yml:22]: You technically skipped all of the reserved fields in all of the MSRs, but this one looks like it really needs a single field defined in this fieldset:

```
- name: Reserved
  long_name: "Reserved"
  lsb: 0
  msb: 63
  readable: True
```

File: data/amd/register/msr/MSRC001_0115.yml [hunk 13]
```diff
@@ -0,0 +1,30 @@
+- name: MSRC001_0115
+  long_name: "Virtual Machine Control"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0115.yml:2]: This should be: `long_name: "IGNNE"`

File: data/amd/register/msr/MSRC001_0130.yml [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+        MSRC001_0130 [Guest Host Communication Block] (Core::X86::Msr::GHCB)
+       
+       "
+  size: 64
+  arch: amd64
+
+  access_mechanisms:
+      - name: rdmsr
+        is_read: True
+        address: 0xC001011E
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0130.yml:14]: This (and the other one below it) should be: `address: 0xC0010130`

File: data/amd/register/msr/MSRC001_0140.yml [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+  fieldsets:
+      - name: latest
+        condition: "Fieldset valid on latest version of the AMD architecture"
+        size: 64
+
+        fields:
+            - name: OSVWIdLength
+              long_name: "OS visible work-around ID length"
+              lsb: 0
+              msb: 0
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0140.yml:28]: This should be `msb: 15`

File: data/amd/register/msr/MSRC001_0141.yml [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+       "
+        MSRC001_0141 [OS Visible Work-around Status] (Core::X86::Msr::OSVW_Status)
+       "
+  size: 64
+  arch: amd64
+
+  access_mechanisms:
+      - name: rdmsr
+        is_read: True
+        address: 0xC0010140
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0141.yml:13]: This (and the one below it) should be: `address: 0xC0010141`

File: data/amd/register/msr/MSRC001_0231.yml [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+       Also can be read via x86 instructions RDPMC ECX=[0F:0A].
+       "
+  size: 64
+  arch: amd64
+  is_indexed: True
+
+  access_mechanisms:
+      - name: rdmsr
+        is_read: True
+        address: 0xC0010201
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0231.yml:16]: This (and the one below it) should be `address: 0xC0010231`

File: data/amd/register/msr/MSRC001_0299.yml [hunk 18]
```diff
@@ -0,0 +1,47 @@
+- name: MSRC001_0299
+  long_name: "RAPL Power Unit"
+  purpose: |
+       "
+        Read-only,Volatile. Reset: 0000_0000_000A_1003h.
+
+        MSRC001_0111 [SMM Base Address] (Core::X86::Msr::SMM_BASE)
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_0299.yml:7]: This description looks wrong

File: data/amd/register/msr/MSRC001_029A.yml [hunk 19]
```diff
@@ -0,0 +1,33 @@
+- name: MSRC001_029A
+  long_name: "Core Energy Status"
+  purpose: |
+       "
+        MSRC001_029A [Core Energy Status] (Core::X86::Msr::CORE_ENERGY_STAT)
+
+        MSRC001_0111 [SMM Base Address] (Core::X86::Msr::SMM_BASE)
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_029A.yml:7]: This part of the description looks wrong

File: data/amd/register/msr/MSRC001_029B.yml [hunk 20]
```diff
@@ -0,0 +1,30 @@
+- name: MSRC001_029B
+  long_name: "Core Energy Status"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_029B.yml:2]: This should be `long_name: "Package Energy Status"`

File: data/amd/register/msr/MSRC001_101A.yml [hunk 21]
```diff
@@ -0,0 +1,32 @@
+- name: MSRC001_101A
+  long_name: "Address Mask For DR1 Breakpoint"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_101A.yml:2]: This should be: `long_name: "Address Mask For DR2 Breakpoint"`

File: data/amd/register/msr/MSRC001_101B.yml [hunk 22]
```diff
@@ -0,0 +1,30 @@
+- name: MSRC001_101B
+  long_name: "Address Mask For DR3 Breakpoint"
+  purpose: |
+       "
+        MSRC001_101A [Address Mask For DR2 Breakpoint] (Core::X86::Msr::DR2_ADDR_MASK)
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_101B.yml:5]: The purpose of this field is wrong, should be: `MSRC001_101B [Address Mask For DR3 Breakpoint] (Core::X86::Msr::DR3_ADDR_MASK)`

File: data/amd/register/msr/MSRC001_1023.yml [hunk 23]
```diff
@@ -0,0 +1,30 @@
+- name: MSRC001_1023
+  long_name: "Address Mask For DR0 Breakpoints"
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_1023.yml:2]: This should be: `long_name: "Table Walker Configuration"`

File: data/amd/register/msr/MSRC001_1033.yml [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+        • An interrupt is generated as specified by Core::X86::Msr::IBS_CTL. The interrupt service routine associated 
+        with this interrupt is responsible for saving the performance information stored in IBS execution registers.
+       "
+  size: 64
+  arch: amd64
+  
+  access_mechanisms:
+      - name: rdmsr
+        is_read: True
+        address: 0xC0011032
```
Comment (JaredWright) [inline data/amd/register/msr/MSRC001_1033.yml:33]: This (and the one below it) needs to be: `address: 0xC0011033`

**Review by JaredWright — CHANGES_REQUESTED** [review]
There are a lot of mistakes in this PR. Please work more carefully when making these files. Getting the value wrong for things like MSR addresses or field msb/lsb values will result in *really* difficult to track down software bugs. Additionally, getting values like "long_name" or "purpose" wrong will result in incorrect labels for PAL's generated "print" functions.

I didn't attempt to verify the accuracy of each individual field definition (things like msb, lsb, etc). In addition to the changes suggested, please go back through and verify that the `msb` and `lsb` values for each field is correct.

File: data/amd/register/msr/MSR0000_01DE.yml [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+        condition: "Fieldset valid on latest version of the AMD architecture"
+        size: 64
+
+        fields:
+            - name: LastIntToIP
+              long_name: "LastIntToIP"
+              lsb: 0
+              msb: 63
+              readable: True 
+              writable: False
```
[the remark this answers is not in the document]
Comment (jaysongiroux) [reply data/amd/register/msr/MSR0000_01DE.yml:33]: MSR0000_0277 is a nested table MSR which will be included in a future PR

MSR0000_02FF has bee included

**Comment by jaysongiroux** [comment]
> There are a lot of mistakes in this PR. Please work more carefully when making these files. Getting the value wrong for things like MSR addresses or field msb/lsb values will result in _really_ difficult to track down software bugs. Additionally, getting values like "long_name" or "purpose" wrong will result in incorrect labels for PAL's generated "print" functions.
> 
> I didn't attempt to verify the accuracy of each individual field definition (things like msb, lsb, etc). In addition to the changes suggested, please go back through and verify that the `msb` and `lsb` values for each field is correct.

Before squashing I pushed two commits: 
1. fixing commit messages
2. I went through each MSR and verified the information and fixed up what was missed or entered incorrectly.

**Pull request merged**